### PR TITLE
wasm: use static registration for runtimes

### PR DESCRIPTION
**Pull request by lizan**
Partially addresses #12574. Refactored test instantiate to removes many ifdefs.

Commit Message:
Additional Description:
Risk Level: Low
Testing: CI
Docs Changes:
Release Notes:
Platform Specific Features:

Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -92,6 +92,8 @@ extensions/filters/common/original_src @snowp @klarose
 /*/extensions/filters/network/wasm @PiotrSikora @lizan
 # webassembly common extension
 /*/extensions/common/wasm @PiotrSikora @lizan
+# webassembly runtimes
+/*/extensions/wasm_runtime/ @PiotrSikora @lizan
```
Comment (PiotrSikora) [inline CODEOWNERS:96]: Any reason for not putting it under `extensions/common/wasm/runtimes` or similar?

**Review by PiotrSikora — COMMENTED** [review]
Looks good, thanks! Could you verify that this works with `--define wasm=disabled` when some of those values evaluate to an empty list? Should we include this config in the CI?

**Comment by lizan** [comment]
> Looks good, thanks! Could you verify that this works with `--define wasm=disabled` when some of those values evaluate to an empty list? Should we include this config in the CI?

Windows CI or Arm64 CI does this :)

File: CODEOWNERS (hunk 1 above)
Comment (lizan) [reply CODEOWNERS:96]: I don't think we put any actual extension registrations under `extensions/common`, `extensions/common` is for libraries used by extensions.

**Comment by PiotrSikora** [comment]
> > Looks good, thanks! Could you verify that this works with `--define wasm=disabled` when some of those values evaluate to an empty list? Should we include this config in the CI?
> 
> Windows CI or Arm64 CI does this :)

Why? V8 should work fine on both Windows and Arm64.

**Review by PiotrSikora — APPROVED** [review]

**Review by mathetake — DISMISSED** [review]
Thank you so much! :slightly_smiling_face:

**Comment by mattklein123** [comment]
Check CI?

/wait

**Review by mattklein123 — APPROVED** [review]
Awesome!

**Pull request merged**